nengo/cache.py

```diff
@@ -105,7 +106,9 @@ def check_seq(tpl):
 
 def check_attrs(obj):
     attrs = [getattr(obj, x) for x in dir(obj) if not x.startswith("_")]
-    return all(Fingerprint.supports(x) for x in attrs if not callable(x))
+    return all(
+        Fingerprint.supports(x) for x in attrs if not callable(x) and x is not None
```

Comment:
When might it be `None`?

Reply:
Whenever an attribute on an object is set to `None`?

Reply:
Yeah, this came up with I think the `Cholesky` solver and maybe a few others that have a `transpose` parameter which by default is `None` (meaning the solver can choose which way to do the computation internally). `Fingerprint` seems to have no problem hashing these. I think it makes more sense to add `NoneType` to the whitelist, though, so I went and did that instead.

Also, this raised the point that the test only tests whether `Fingerprint.supports` is true for the type. I added a second part to the test to make sure it actually hashes without a problem (just in case something gets on the whitelist that isn't hashable for some reason).